### PR TITLE
Skip non ref version

### PR DESCRIPTION
**Pull request by corest**
## Checklist

- [x] Update changelog in CHANGELOG.md.

My release action here failed https://github.com/giantswarm/cert-operator/actions/runs/203867180
Validation is correct for this case, also we skip setting a new version as we need to keep the same version

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+- Skip vaidation of reference versions like `0.1.0-1`.
```
Comment (stone-z) [inline CHANGELOG.md:12]:
```suggestion
- Skip validation of reference versions like `0.1.0-1`.
```

**Comment by corest** [comment]
I would rather skip this workflow completely for replacement releases as it starts to be overcomplicated for this rare case. But can we skip the whole action based on a condition?

File: .github/workflows/zz_generated.create_release.yaml [hunk 2]
```diff
@@ -118,8 +118,15 @@ jobs:
           echo "::add-path::/opt/bin"
       - name: Checkout code
         uses: actions/checkout@v2
+      - name: Ensure non-reference version
+        id: non_ref-version
```
Comment (kopiczko) [inline .github/workflows/zz_generated.create_release.yaml:122]:
```suggestion
        id: non_ref_version
```

File: .github/workflows/zz_generated.create_release.yaml [hunk 3]
```diff
@@ -118,8 +118,15 @@ jobs:
...
+        run: |
+          if [[ "${{ needs.gather_facts.outputs.version }}" =~ ^[0-9]+.[0-9]+.[0-9]+-[0-9]+$ ]]; then
```
Comment (kopiczko) [inline .github/workflows/zz_generated.create_release.yaml:124]: How about adding a new output like `update_project_go: true/false` in `gather_facts` and skipping the whole job (`update_project_go`) if it's set to false?

**Comment by kopiczko** [comment]
> I would rather skip this workflow completely for replacement releases as it starts to be overcomplicated for this rare case. But can we skip the whole action based on a condition?

Tag and changelog should be updated.

File: .github/workflows/zz_generated.create_release.yaml (hunk 3 above)
Comment (corest) [reply .github/workflows/zz_generated.create_release.yaml]: It is used also in the validation before tagging so I keep it as `refversion`

File: .github/workflows/zz_generated.create_release.yaml [hunk 4]
```diff
@@ -59,6 +59,13 @@ jobs:
           fi
           echo "path=\"$path\""
           echo "::set-output name=path::${path}"
+      - name: Check if reference version
+        id: ref_version
+        run: |
+          version=$(echo $title | cut -d ' ' -f 2)
+          if [[ "${version}" =~ ^[0-9]+.[0-9]+.[0-9]+-[0-9]+$ ]]; then
+            echo ::set-output name=refversion::true
+          fi
```
Comment (kopiczko) [inline .github/workflows/zz_generated.create_release.yaml:68]:
```suggestion
          refversion=false
          if [[ "${version}" =~ ^[0-9]+.[0-9]+.[0-9]+-[0-9]+$ ]]; then
            refversion=true
          fi
          echo "refversion =\"$refversion\""
          echo "::set-output name=refversion::$refversion"
```

**Pull request merged**